javascript/javascript2/week2/hyf-bay/main.js

```diff
@@ -0,0 +1,74 @@
+console.log("Script loaded");
+
+const products = getAvailableProducts();
+console.log(products);
+let productsList = document.querySelector("main ul");
+function renderProducts(products) {
+        products.forEach(element => {
+        let items = document.createElement("li");
+        let itemName = document.createElement("h3");
+        let price = document.createElement("p");
+        let rating = document.createElement("p");
+        itemName.innerHTML = element.name;
+        price.innerHTML = `Price: ${element.price}`;
+        rating.innerHTML = `Rating: ${element.rating}`;
+        items.append(itemName, price, rating);
+        productsList.append(items);
+    });
+}
+
+renderProducts(products);
+
+
+//Search by name
+let searchInput = document.getElementById("search-filter");
+
+searchInput.addEventListener("keyup", searching);
+let filteredProductsByName;
+function searching(){
+    let searchValue = searchInput.value;
+    productsList.innerHTML = "";
+    filteredProductsByName = products.filter(item => item.name.toLowerCase().includes(searchValue.toLowerCase()));
+    renderProducts(filteredProductsByName);
+}
+
+//Search by max price
+let maxPriceInput = document.getElementById("max-price-filter");
+let filteredProductsByMaxPrice;
+maxPriceInput.addEventListener("keyup", maxPriceFilter)
+function maxPriceFilter(){
+    let maxPriceValue = maxPriceInput.value;
+    productsList.innerHTML = "";
+    filteredProductsByMaxPrice = filteredProductsByName ? filteredProductsByName.filter(item => item.price < maxPriceValue) : products.filter(item => item.price < maxPriceValue);
```

Comment:
It is for the case when using also search filter for looking up items by their name at the same time.
FilteredProducts by name is undefined as long as the search bar is not used, for that case, it uses the complete array products  to filter.
In case the search bar is also used(for look items by their name), then FilteredProducts is defined and it is a shorter array of the items previously filtered by the name.